shcDecoder/src/main/java/ca/bc/gov/shcdecoder/cache/impl/CacheManagerImpl.kt

```diff
@@ -17,27 +19,50 @@ internal class CacheManagerImpl(
     companion object {
         const val SUFFIX_JWKS_JSON = "/.well-known/jwks.json"
         const val SUFFIX_ISSUER_JSON = "issuers.json"
-        private const val TAG = "CacheManagerImpl"
     }
 
     override suspend fun fetch() {
         if (isCacheExpired()) {
-            try {
-                fileManager.downloadFile(shcConfig.issuerEndPoint)
-                val issuers = fileManager.getIssuers(shcConfig.issuerEndPoint)
-                issuers.forEach { issuer ->
-                    val keyUrl = if (issuer.iss.endsWith(SUFFIX_JWKS_JSON)) {
-                        issuer.iss
-                    } else {
-                        "${issuer.iss}$SUFFIX_JWKS_JSON"
-                    }
-                    fileManager.downloadFile(keyUrl)
-                }
+            fileManager.downloadFile(shcConfig.rulesEndPoint)
```

Comment:
Please wrap this inside try-catch block as we don't want to block download & cache for other URL if one of them fails.

Reply:
the downloadFile method code is wrapped in a try catch block, in theory it should not be possible for this to block others. What do you think?